src/dmd/lambdacomp.d

```diff
@@ -108,35 +154,43 @@ extern (C++) class SerializeVisitor : SemanticTimeTransitiveVisitor
         if (buf.offset == 0)
             return;
 
+        // The identifier may be an argument
         auto id = exp.ident.toChars;
         auto stringtable_value = arg_hash.lookup(id, strlen(id));
         if (stringtable_value)
         {
+            // In which case we need to update the serialization accordingly
             const(char)* gen_id = cast(const(char)*)stringtable_value.ptrvalue;
             buf.writestring(gen_id);
             buf.writeByte('_');
             et = ExpType.Arg;
         }
         else
         {
+            // Or it may be something else
             Dsymbol scopesym;
             Dsymbol s = sc.search(exp.loc, exp.ident, &scopesym);
             if (s)
             {
+                // If it's a VarDeclaration
                 if (auto v = s.isVarDeclaration)
                 {
+                    // It must be a manifest constant
                     if (v.storage_class & STC.manifest)
                     {
                         v.getConstInitializer.accept(this);
                     }
+                    // Otherwise, the function is uncomparable
```

Comment:
This flow is a little confusing.
please let the buf.reset be a fall-through case.

Reply:
it is a fall-through case, I just need it to be reset inside the first if-block if the var declaration isn't a manifest constant. I don't know now to make differently without adding a label, which in my opinion would be worse.

Reply:
it could also be auto v = s.isVarDeclaration if (v && v.storage_class & STC.manifest)

In general if (a) {if (b) {c}} maps exactly to if (a && b) {c}